hack/generated/pkg/reconcilers/azure_deployment_reconciler.go

```diff
@@ -593,90 +606,39 @@ func (r *AzureDeploymentReconciler) MonitorDeployment(ctx context.Context) (ctrl
+	// TODO: Create a unit test that forces this specific sequence of events
+	r.log.V(4).Info("Deleting deployment", "DeploymentID", deployment.ID)
+	_, err = r.ARMClient.DeleteDeployment(ctx, deployment.ID)
 	if err != nil {
-		// This is a superfluous error as per https://github.com/kubernetes-sigs/controller-runtime/issues/377
-		// The correct handling is just to ignore it and we will get an event shortly with the updated version to patch
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+		return ctrl.Result{}, errors.Wrapf(err, "failed deleting deployment %q", deployment.ID)
 	}
 
-	// TODO: Could somehow have a method that grouped both of these calls
-	currentState := r.GetResourceProvisioningState()
-	r.log.V(4).Info("Monitoring deployment", "action", string(CreateOrUpdateActionMonitorDeployment), "id", deployment.ID, "state", currentState)
-	r.recorder.Event(r.obj, v1.EventTypeNormal, string(CreateOrUpdateActionMonitorDeployment), fmt.Sprintf("Monitoring Azure deployment ID=%q, state=%q", deployment.ID, currentState))
-
-	// We do two patches here because if we remove the deployment before we've actually confirmed we persisted
-	// the resource ID, then we will be unable to get the resource ID the next time around. Only once we have
-	// persisted the resource ID can we safely delete the deployment
-	retryAfter = time.Duration(0) // ARM can tell us how long to check after issuing DELETE
-	if deployment.IsTerminalProvisioningState() && !r.GetShouldPreserveDeployment() {
-		r.log.Info(
-			"Deployment in terminal state",
-			"DeploymentID", deployment.ID,
-			"State", deployment.ProvisioningStateOrUnknown(),
-			"Error", deployment.ErrorOrEmpty())
-
-		r.log.V(4).Info("Deleting deployment", "DeploymentID", deployment.ID)
-
-		retryAfter, err = r.ARMClient.DeleteDeployment(ctx, deployment.ID)
-		if err != nil {
-			return ctrl.Result{}, errors.Wrapf(err, "failed deleting deployment %q", deployment.ID)
-		}
-
-		deployment.ID = ""
-		deployment.Name = ""
-
-		err = r.Update(deployment, status)
-		if err != nil {
-			return ctrl.Result{}, errors.Wrap(err, "updating obj")
-		}
-
-		err = r.CommitUpdate(ctx)
-		if err != nil {
-			// This is a superfluous error as per https://github.com/kubernetes-sigs/controller-runtime/issues/377
-			// The correct handling is just to ignore it and we will get an event shortly with the updated version to patch
-			return ctrl.Result{}, client.IgnoreNotFound(err)
-		}
-
-		r.log.V(4).Info(
-			"Cleared deployment info from resource",
-			"DeploymentID", r.GetDeploymentIDOrDefault(),
-			"DeploymentName", r.GetDeploymentNameOrDefault())
-	}
-
-	if deployment.IsTerminalProvisioningState() {
-		// we are done
-		return ctrl.Result{}, nil
-	}
+	deployment.ID = ""
+	deployment.Name = ""
 
-	r.log.V(3).Info("Deployment still running")
-	return ctrl.Result{Requeue: true, RequeueAfter: retryAfter}, err
+	// TODO: Do people prefer this or just putting this method body inline
```

Comment:
Thoughts on this?

Reply:
I created a method that is only used in a single place and was second guessing whether it was just cleaner to do it inline in this method. That makes this method longer though...

I've just removed the TODO and kept the `handleDeploymentFinished` for now. We can always change it later if we don't like it.

Reply:
A single use method is often still a good idea because it gives that piece of functionality a name, and allows consideration in isolation.

Reply:
Yeah that was my thought when I wrote it, but sometimes you can also overuse it and end up in really deep method chains. For some reason I was hesitant about it when I wrote it but in reviewing the PR it seems just net-good.